src/directives/formly-field.js

```diff
@@ -87,9 +87,9 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
       if (!key || !model) {
         return
       }
+
       // If we are working with a number then $parse wont work, default back to the old way for now
-      if (angular.isNumber(key)) {
-        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
+      if (angular.isNumber(key) || (/\d/.test(key) && key.indexOf('[') === -1)) {
```

Comment:
Do you suppose that `/\d/` could actually be `/^\d/`? Isn't the number problem only present if the first character of the key is a number? I feel like if we implement this as it is, we'd have a bug where a key like this: `w390ekd032s.waky` would result in the model getting the property set incorrectly. (same for the test below).

Actually, also, it'd probably be nice to pull this if statement out into a function that both this line and the one below can use to avoid bugs in the future (changing one but not the other)